src/components/Nav/Menu/index.tsx

```diff
@@ -0,0 +1,107 @@
+import { AnimatePresence, motion } from "framer-motion"
+import { Box, type BoxProps, Flex, Text } from "@chakra-ui/react"
+import * as NavigationMenu from "@radix-ui/react-navigation-menu"
+
+import { Button } from "@/components/Buttons"
+
+import { SECTION_LABELS } from "@/lib/constants"
+
+import type { NavSections } from "../types"
+
+import SubMenu from "./SubMenu"
+import { useNavMenu } from "./useNavMenu"
+
+type NavMenuProps = BoxProps & {
+  sections: NavSections
+}
+
+const Menu = ({ sections, ...props }: NavMenuProps) => {
+  const {
+    activeSection,
+    containerVariants,
+    direction,
+    handleSectionChange,
+    isOpen,
+    menuColors,
+    onClose,
+  } = useNavMenu(sections)
+
+  return (
+    <Box {...props}>
+      <NavigationMenu.Root
+        dir={direction}
```

Comment:
Nice. So it was mostly what you needed to do in terms of rtl?

Reply:
Yea, this seems to have been enough for it to propagate down the tree to the sub-menus.